### PR TITLE
Code reference in help

### PR DESCRIPTION
**Pull request by rabuzarus**
This gives the possibility to have a code reference (documentation of classes and functions) in the help page.

The reference can be built with (doxygen needs to be installed):
`doxygen doxygen util/Doxyfile'`

The generated docu is under doc/html

This gives a little reference to the developers about friendicas code and would help newcomers. Much code is undocumented at the present time (or the doxygen format isn't used). I hope this could change with the time.

### CODE REVIEW

File: util/Doxyfile [hunk 1]
```diff
@@ -0,0 +1,23 @@
+INPUT = README.md index.php boot.php testargs.php update.php mod/ object/ include/ js/ util/ view/ version.inc
+RECURSIVE = YES
+PROJECT_NAME = "Friendica"
+PROJECT_LOGO = images/friendica-64.jpg
+EXCLUDE = .htconfig.php library/ doc/ .git/ log/ util/zotsh/easywebdav/
```
Comment (annando) [inline util/Doxyfile]: Could you have a look in the .gitignore for all folders that are excluded there to include them here as well?

**Comment by rabuzarus** [comment]
ok I added some folders.
should I add the folders of the IDE as well to exclude?

File: view/theme/boheme [hunk 2]
```diff
@@ -0,0 +1 @@
+Subproject commit 3009bdd2c2720d481028cb03671a5987160b8279
```
Comment (fabrixxm) [inline view/theme/boheme]: this shouldn't be here, I think...

Comment (rabuzarus) [inline view/theme/boheme]: Ahhh... Why is it there? And more important how can I delete it?

**Comment by annando** [comment]
Which IDE?

File: .gitignore [hunk 3]
```diff
@@ -42,3 +42,12 @@ nbproject
 
 #ignore local folder
 /local/
+
+.directory
```
Comment (annando) [inline .gitignore]: What is this ".directory" for?

**Comment by rabuzarus** [comment]
Sorry all this commit confusion, I updated netbeans yesterday. And now it restores the .gitignore everytime when I change the repository.

According to IDE. In gitignore is:
#ignore config files from eclipse, we don't want IDE files in our repository
.project
.buildpath
.externalToolBuilders
.settings
#ignore OSX .DS_Store files 
.DS_Store

/nbproject/private/

**Pull request merged**